### PR TITLE
feat(auth): Implemented GeneratePasswordResetLinkAsync() API

### PR DESCRIPTION
**Pull request by hiranya911**
API for email action link generation. API proposal go/firebase-dotnet-mt

```
class ActionCodeSettings
  string Url
  boolean HandleCodeInApp
  string DynamicLinkDomain
  string IosBundleId
  string AndroidPackageName
  string AndroidMinimumVersion
  boolean AndroidInstallApp
class FirebaseAuth
  Task<string> GeneratePasswordResetLinkAsync(string email)
  Task<string> GeneratePasswordResetLinkAsync(string email, ActionCodeSettings settings)
```

### CODE REVIEW

**Comment by hiranya911** [comment]
@egilmorez please review the API docs in `ActionCodeSettings` and `FirebaseAuth` classes.

File: FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs [hunk 1]
```diff
@@ -552,6 +552,57 @@ await this.SetCustomUserClaimsAsync(uid, claims, default(CancellationToken))
...
+                .ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Generates the out-of-band email action link for password reset flows for the specified
+        /// email address.
+        /// </summary>
+        /// <exception cref="FirebaseAuthException">If an error occurs while setting custom claims.</exception>
+        /// <param name="email">The email of the user whose password is to be reset.</param>
+        /// <param name="settings">The action code settings object which defines whether
```
Comment (egilmorez) [inline FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs:574]: I think this technically should be "that" -- you're referring to a particular object that has these attributes, and not just adding information about the object.

File: FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs [hunk 2]
```diff
@@ -552,6 +552,57 @@ await this.SetCustomUserClaimsAsync(uid, claims, default(CancellationToken))
...
+        /// the link is to be handled by a mobile app and the additional state information to be
+        /// passed in the deep link..</param>
```
Comment (egilmorez) [inline FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs:576]: Extra period here.

File: FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs [hunk 3]
```diff
@@ -552,6 +552,57 @@ await this.SetCustomUserClaimsAsync(uid, claims, default(CancellationToken))
...
+                .ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Generates the out-of-band email action link for password reset flows for the specified
+        /// email address.
+        /// </summary>
+        /// <exception cref="FirebaseAuthException">If an error occurs while setting custom claims.</exception>
+        /// <param name="email">The email of the user whose password is to be reset.</param>
+        /// <param name="settings">The action code settings object which defines whether
```
Comment (egilmorez) [inline FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs:591]: Suggest "that," as above.

File: FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs [hunk 4]
```diff
@@ -552,6 +552,57 @@ await this.SetCustomUserClaimsAsync(uid, claims, default(CancellationToken))
...
+        /// the link is to be handled by a mobile app and the additional state information to be
+        /// passed in the deep link..</param>
```
Comment (egilmorez) [inline FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs:593]: Extra period.

File: FirebaseAdmin/FirebaseAdmin/Auth/ActionCodeSettings.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        /// </item>
+        /// </list>
+        /// This property is required.
+        /// </summary>
+        public string Url { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to open the link via a mobile app or a browser.
+        /// The default is false. When set to true, the action code link is sent as a Universal
+        /// Link or an Android App Link and is opened by the app if installed. In the false case,
```
Comment (egilmorez) [inline FirebaseAdmin/FirebaseAdmin/Auth/ActionCodeSettings.cs:47]: Is any of this relevant when the app isn't installed?  I wonder if we need this phrase.

File: FirebaseAdmin/FirebaseAdmin/Auth/ActionCodeSettings.cs [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+        /// <summary>
+        /// Gets or sets the Android package name of the app where the link should be handled if
+        /// the Android app is installed. Must be specified when setting other Android-specific
+        /// settings.
+        /// </summary>
+        public string AndroidPackageName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the minimum version for Android app. If the installed app is an older
```
Comment (egilmorez) [inline FirebaseAdmin/FirebaseAdmin/Auth/ActionCodeSettings.cs:74]: Suggest "an Android app," or "the" if we mean the specific one being targeted.

File: FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs (hunk 4 above)
Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs:593]: Done

File: FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs (hunk 3 above)
Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs:591]: Done

File: FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs (hunk 2 above)
Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs:576]: Done

File: FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs (hunk 1 above)
Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin/Auth/FirebaseAuth.cs:574]: Done

File: FirebaseAdmin/FirebaseAdmin/Auth/ActionCodeSettings.cs (hunk 6 above)
Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin/Auth/ActionCodeSettings.cs:74]: Done

File: FirebaseAdmin/FirebaseAdmin/Auth/ActionCodeSettings.cs (hunk 5 above)
Comment (hiranya911) [reply FirebaseAdmin/FirebaseAdmin/Auth/ActionCodeSettings.cs:47]: I think those are orthogonal issues. If the app isn't installed, you will still get a dynamic link that will offer to install the app.

**Pull request merged**